pkg/xray/resource_xray_watch.go

```diff
@@ -0,0 +1,118 @@
+package xray
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func resourceXrayWatch() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceXrayWatchCreate,
+		ReadContext:   resourceXrayWatchRead,
+		UpdateContext: resourceXrayWatchUpdate,
+		DeleteContext: resourceXrayWatchDelete,
+		Description:   "Provides an Xray watch resource.",
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Name of the watch (must be unique)",
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Description of the watch",
+			},
+			"active": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "Whether or not the watch will be active",
+			},
+			"watch_resource": {
+				Type:        schema.TypeList,
+				Required:    true,
+				Description: "Nested argument describing the resources to be watched. Defined below.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"type": {
+							Type:             schema.TypeString,
+							Required:         true,
+							Description:      "Type of resource to be watched. Options: `all-repos`, `repository`, `build`, `project`, `all-projects`.",
+							ValidateDiagFunc: validation.ToDiagFunc(validation.StringInSlice([]string{"all-repos", "repository", "build", "project", "all-projects"}, true)),
+						},
+						"bin_mgr_id": {
+							Type:        schema.TypeString,
+							Optional:    true,
+							Default:     "default",
+							Description: "The ID number of a binary manager resource. Should be set to `default` if not set on the Artifactory side.",
+						},
+						"name": {
+							Type:        schema.TypeString,
+							Optional:    true,
+							Description: "The name of the build or repository. Enable Xray indexing must be enabled on the repo or build",
+						},
+						"filter": {
+							Type:        schema.TypeList,
+							Optional:    true,
```

Comment:
Even if it's a regex, you can/should still validate it - and, does a list make sense?

Reply:
"value":{"ExcludePatterns":[],"IncludePatterns":["*"]} is still not supported. What should I validate it against?

Reply:
What do you mean, it's not supported?

Reply:
I've created a ticket for that, it will be done later.